### PR TITLE
Beyla gRPC support for the sample

### PR DESCRIPTION
**Pull request by dashpole**
Also, migrate to the OTel conventions, as they are supported in the in-context dashboard now.

The one caveat is that the in-context dashboard looks for `rpc_server_duration_count` instead of `rpc_server_duration_seconds_count`, so remove the unit to keep the unit suffix out.

Some gRPC metrics also seem to have an empty rpc.method field, which i've filtered out.
Screenshot 2024-05-24 at 4 56 11 PM

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

**Comment by dashpole** [comment]
The service graph works, but beyla seems to be missing requests from the frontend to other applications.  I checked the traces we were getting for the frontend, where the frontend is the client, and there were none.

**Review by aabmass — APPROVED** [review]

File: recipes/beyla-golden-signals/collector-config.yaml [hunk 1]
```diff
@@ -39,46 +39,29 @@ spec:
...
-            - name != "http.server.request.duration"
-      # Transform metrics to match GMP http server conventions:
-      # https://cloud.google.com/stackdriver/docs/managed-prometheus/exporters/server/http
-      transform/prometheusconventions:
+            - name != "http.server.request.duration" and name != "rpc.server.duration"
+          datapoint:
+            # Some gRPC metrics appear not to have an RPC method.
+            - metric.name == "rpc.server.duration" and attributes["rpc.method"] == ""
+            # Some HTTP metrics appear not to have an http method.
+            - metric.name == "http.server.request.duration" and attributes["http.request.method"] == ""
```
[the remark this answers is not in the document]
Comment (dashpole) [reply recipes/beyla-golden-signals/collector-config.yaml:47]: The graph will show a series without any method, which seems wrong.  I do suspect it is a beyla bug, and i'm looking into it

**Comment by dashpole** [comment]
I'm not writing to the gRPC conventions, rather than the OTel RPC conventions to make the latency units correct.

**Comment by dashpole** [comment]
Ok, final change.  I'm now writing the beyla metrics as-is.  I'm going to update the in-context dashboard to support the existing beyla metrics.

**Pull request merged**